src/Middleware/HttpLogging/src/HttpLoggingMiddleware.cs

```diff
@@ -50,88 +54,115 @@ public Task Invoke(HttpContext context)
             return _next(context);
         }
 
-        return InvokeInternal(context);
+        var options = _options.CurrentValue;
+        var loggingAttribute = context.GetEndpoint()?.Metadata.GetMetadata<HttpLoggingAttribute>();
+        var loggingFields = loggingAttribute?.LoggingFields ?? options.LoggingFields;
+
+        if (_interceptors.Length == 0 && loggingFields == HttpLoggingFields.None)
+        {
+            // Logging is disabled for this endpoint and there are no interceptors to turn it on.
+            return _next(context);
+        }
+
+        return InvokeInternal(context, options, loggingAttribute, loggingFields);
     }
 
-    private async Task InvokeInternal(HttpContext context)
+    private async Task InvokeInternal(HttpContext context, HttpLoggingOptions options,
+        HttpLoggingAttribute? loggingAttribute, HttpLoggingFields loggingFields)
     {
-        var options = _options.CurrentValue;
         RequestBufferingStream? requestBufferingStream = null;
         Stream? originalBody = null;
 
-        var loggingAttribute = context.GetEndpoint()?.Metadata.GetMetadata<HttpLoggingAttribute>();
-        var loggingFields = loggingAttribute?.LoggingFields ?? options.LoggingFields;
+        var logContext = _contextPool.Get();
+        logContext.HttpContext = context;
+        logContext.LoggingFields = loggingFields;
+        logContext.RequestBodyLogLimit = options.RequestBodyLogLimit;
+        logContext.ResponseBodyLogLimit = options.ResponseBodyLogLimit;
+        logContext.StartTimestamp = _timeProvider.GetTimestamp();
+        logContext.TimeProvider = _timeProvider;
 
-        if ((HttpLoggingFields.Request & loggingFields) != HttpLoggingFields.None)
+        if (loggingAttribute?.IsRequestBodyLogLimitSet is true)
+        {
+            logContext.RequestBodyLogLimit = loggingAttribute.RequestBodyLogLimit;
+        }
+        if (loggingAttribute?.IsResponseBodyLogLimitSet is true)
         {
-            var request = context.Request;
-            var list = new List<KeyValuePair<string, object?>>(
-                request.Headers.Count + DefaultRequestFieldsMinusHeaders);
+            logContext.ResponseBodyLogLimit = loggingAttribute.ResponseBodyLogLimit;
+        }
 
+        for (var i = 0; i < _interceptors.Length; i++)
+        {
+            await _interceptors[i].OnRequestAsync(logContext);
```

Comment:
Does this code need to be in try/finally in order to return the `LogContext` to the pool?